app/controllers/client_data_controller.rb

```diff
@@ -94,12 +94,20 @@ def auth_errors(exception)
     path = polymorphic_path(model_class, action: :new)
     # prevent redirect loop
     if path == request.path
-      render "authorization_error", status: 403, locals: { msg: exception.message }
+      render_auth_errors(exception)
     else
       redirect_to path, alert: exception.message
     end
   end
 
+  def render_auth_errors(exception)
+    if exception.message == "Client is disabled"
+      render_disabled_client_message(exception.message)
```

Comment:
yes, it would currently. but it would break encapsulation to assume that an exception-specific rendering method (`render_disabled_client_message`) would always use the same template to render as a generic auth exception.

It's a subtle difference, I realize, but consider this scenario. Six months from now we have another need for a disabled client, only this time we have a UI paradigm that has a specific "disabled" template we want to use as a fallback for all clients.

So six-months-from-now you says, oh, I'll just change the default fallback template and all will be well:
```diff
  def render_disabled_client_message(message)
    begin
      render "#{current_user.client_slug}/_disabled", status: 403
    rescue ActionView::MissingTemplate => _error
-      render "authorization_error", status: 403, locals: { msg: message }
+     render "custom_disabled_template", status: 403, locals: { msg: message }
     end
   end
```

and lo! you've now introduced a regression where every auth error renders using that "custom_disabled_template".

That's why I wrote it like I did: encapsulating the template-picking logic within the method based on what the exception is, not what the template is.